uimaj-parent/pom.xml

```diff
@@ -396,10 +396,49 @@
                 </configuration>
               </execution>
             </executions>
-            
           </plugin>
         </plugins>
       </build>
     </profile>
+    <profile>
```

Comment:
Why is this in a profile?  Why not just add a pluginExecutionFilter config to the existing lifecycle-mapping in the uimaj-parent pom?

Reply:
No need for it to be part of the build unless the build is running in an M2E environment. If m2e is detected via the system property, then the profile activates. I find this nicer as it keeps context-specific configurations nicely compartmentalised.

Reply:
Sorry, I didn't provide enough information, perhaps.  These lifecycle mappings for m2e are never part of the (non-Eclipse) builds - they're ignored.  They only have an effect for eclipse m2e implementations, as far as I know.  So the profile is not needed, to achieve the effect of not having it be part of the build when not running in the m2e environment.

Also, the testing of the property "m2e.version" to activate the profile: how did you determine that this property would be set in an m2e environment (and not otherwise)?  Where does it come from? where is this documented?  It seems mysterious to me (but, of course, perhaps I've forgotten / overlooked something "obvious" :-) ).

Reply:
To be honest, I have no idea where I picked up this property. I've been using it in DKPro Core for probably 10 years or so. But probably I used it even before that when I was looking for a way to execute certain Maven goals during Eclipse builds. If you google for `property "m2e.version"` you'll find several mailing list posts and other sources mentioning it across the web.

Actually, the plugin is not ignored - it is used but it doesn't do anything. Putting it into a profile that is not activated when it is not needed effectively ignores it.

But honestly, I'm not going to get religious about putting this into a profile or not. You are right: the profile is not required to achieve the desired effect. I still find it nicer because the POM tends to be littered with configurations with some of them belonging together but them not being "scoped" together in some way.

Reply:
Several sources point to this URL in relation to the `m2e.version` property:

https://docs.sonatype.org/display/M2ECLIPSE/Customizable+build+lifecycle+mapping+for+m2e+extensions+developers

However, the website no longer exists and not even the Wayback machine seems to hold a copy...

Reply:
Well, I'm beginning to think I got confused... it seems that the v3 version already was using this profile activation approach!  and checking the v2 master, (before any of this work), it seems to be completely missing the lifecycle-mapping configuration.

Ah-Ha!  it seems bit of config was moved into the uima-wide parent pom.  In particular, https://issues.apache.org/jira/browse/UIMA-3047 talks about moving this into the special profile.

So, I think we should a) cancel this fix to the master-v2, b) update instead the existing uima-wide build (which is still in svn...), with these two improvements (thanks for finding them, Richard! :-) ), and c) remove from v3 the uimaj-parent pom sections related to this - once we release the next version of the uima-wide build.   WDYT?

Reply:
Fine by me. I also have made these changes in the uimaFIT POM and would remove them from there as well once the parent POM has this.

Reply:
ok, I can do this in the uima-wide build pom - and do a release...  or you're welcom to do it :-)  Do you have a preference?

Reply:
Well, I see you've done it already :-)  Thanks!  I can do the abbreviated "release" process for this, or do you want to do that?